Pod/Classes/VOKCoreDataManager.m

```diff
@@ -171,47 +147,74 @@ - (void)initPersistentStoreCoordinator
         storeType = NSSQLiteStoreType;
     }
     
-    if (![_persistentStoreCoordinator addPersistentStoreWithType:storeType
-                                                   configuration:nil
-                                                             URL:storeURL
-                                                         options:options
-                                                           error:&error]) {
-        if (self.migrationFailureOptions == VOKMigrationFailureOptionWipeRecoveryAndAlert) {
-            [[[UIAlertView alloc] initWithTitle:@"Migration Failed"
-                                        message:@"Migration has failed, data will be erased to ensure application stability."
-                                       delegate:nil
-                              cancelButtonTitle:@""
-                              otherButtonTitles:nil] show];
-        }
-        
-        if (self.migrationFailureOptions == VOKMigrationFailureOptionWipeRecoveryAndAlert ||
-            self.migrationFailureOptions == VOKMigrationFailureOptionWipeRecovery) {
-            VOK_CDLog(@"Full database delete and rebuild");
-            [[NSFileManager defaultManager] removeItemAtPath:storeURL.path error:nil];
-            if (![_persistentStoreCoordinator addPersistentStoreWithType:storeType
-                                                           configuration:nil
-                                                                     URL:storeURL
-                                                                 options:nil
-                                                                   error:&error]) {
-                [NSException raise:@"Vokoder Persistant Store Creation Failure after migration"
-                            format:@"Unresolved error %@, %@", error, [error userInfo]];
+    NSError *error;
+    
+    if (![persistentStoreCoordinator addPersistentStoreWithType:storeType
+                                                  configuration:nil
+                                                            URL:storeURL
+                                                        options:options
+                                                          error:&error]) {
+        switch (self.migrationFailureOptions) {
+            case VOKMigrationFailureOptionWipeRecoveryAndAlert:
+            {
+                NSString *title = @"Migration Failed";
+                NSString *message = @"Migration has failed, data will be erased to ensure application stability.";
+#ifdef __IPHONE_8_0 //if compiling with an old version of Xcode that doesn't include the iOS 8 SDK, ignore UIAlertController
+                if ([UIAlertController class]) {
```

Comment:
Yes, otherwise you'd crash when running on iOS < 8.  E.g.: I have an app that I compile on Xcode 6 that is built against the iOS 8 SDK but supports iOS 7 too.